tests/terraform/graph/graph_builder/test_graph_builder.py

```diff
@@ -372,6 +373,13 @@ def test_build_rustworkx_graph(self):
         self.check_edge(graph, resource_node, var_region_node, 'region')
         self.check_edge(graph, provider_node, var_aws_profile_node, 'profile')
         self.check_edge(graph, local_node, var_bucket_name_node, 'bucket_name')
+    
+    def test_multiple_modules_with_connected_resources(self):
+        valid_plan_path = os.path.realpath(os.path.join(TEST_DIRNAME, '../resources/modules_edges_tfplan/tfplan.json'))
+        definitions, definitions_raw = create_definitions(root_folder=None, files=[valid_plan_path])
+        graph_manager = TerraformGraphManager(db_connector=RustworkxConnector())
+        tf_plan_local_graph = graph_manager.build_graph_from_definitions(definitions, render_variables=False)
+        self.assertTrue(tf_plan_local_graph.in_edges[2])
```

Comment:
@sourava01 Can please clarify the test assertion?
1. Are you checking there is an existing in_edges in this index?
2. I think it would be safer to check that a specific edge is created (by the unique resource that should be there with the change) 🙏

Reply:
@SteveVaknin thanks for your reply!
The test case checks if the edges are mapped to the correct resource or not.
When we create 2 or more modules of the same type, the edges get created successfully.

However, all edges get mapped to a single resource only, resulting in false errors for rest of the resources.

Example Scenario - 
`tests/terraform/graph/resources/modules_edges_tfplan/s3module.tf`
This file creates 4 resources:
R0: `module.s3-bucket-1.aws_s3_bucket[0]`
R1: `module.s3-bucket-1.aws_s3_bucket_public_access_block[0]`
R2: `module.s3-bucket-2.aws_s3_bucket[0]`
R3: `module.s3-bucket-2.aws_s3_bucket_public_access_block[0]`

Checkov creates 8 edges in total. 
Expected: 4 edges from R1 to R0 and 4 edges from R3 to R2
Observed: 4 edges from R1 to R0 and 4 edges from R3 to R0

R3 to R0 mapping is incorrect. Edge gets created but checkov cannot find the right vertex which should be attached.

So, through this test case I want to ensure that edges are mapped to the correct resources. 

Not sure how else we can check this. Please lmk if you have any suggestions.

Reply:
Got it, sounds good to me 🚀